daemon/src/main/java/org/sonarlint/daemon/model/ProxyIssueListener.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SonarLint Daemon
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.daemon.model;
+
+import org.sonarsource.sonarlint.core.client.api.common.analysis.ClientInputFile;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.IssueListener;
+import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.Issue;
+import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.Issue.Severity;
+
+import io.grpc.stub.StreamObserver;
+
+public class ProxyIssueListener implements IssueListener {
+  private final StreamObserver<Issue> observer;
+
+  public ProxyIssueListener(StreamObserver<Issue> observer) {
+    this.observer = observer;
+  }
+
+  @Override
+  public void handle(org.sonarsource.sonarlint.core.client.api.common.analysis.Issue issue) {
+    Severity severity;
+    ClientInputFile inputFile = issue.getInputFile();
+
+    switch (issue.getSeverity()) {
+      case "MINOR":
+        severity = Severity.MINOR;
+        break;
+      case "BLOCKER":
+        severity = Severity.BLOCKER;
+        break;
+      case "INFO":
+        severity = Severity.INFO;
+        break;
+      case "CRITICAL":
+        severity = Severity.CRITICAL;
+        break;
+      case "MAJOR":
+      default:
+        severity = Severity.MAJOR;
+        break;
+    }
+
+    Issue.Builder builder = Issue.newBuilder();
+    builder.setRuleKey(issue.getRuleKey())
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1067)